packages/nx/src/command-line/migrate.ts

```diff
@@ -487,72 +488,99 @@ function versions(root: string, from: Record<string, string>) {
 
 // testing-fetch-start
 function createFetcher() {
-  const migrationsCache: Record<
-    string,
-    Promise<ResolvedMigrationConfiguration>
-  > = {};
-  const resolvedVersionCache: Record<string, Promise<string>> = {};
+  const migrationsCache: Record<string, ResolvedMigrationConfiguration> = {};
+  const resolvedVersionCache: Record<string, string> = {};
```

Comment:
Is there a reason for changing these away from the Promise<T> signatures? IIRC there was a reason behind it.

Reply:
Overall my problem with the current implementation is that the flow is very hard to follow. The `setCache` method relies on some side-effect since it doesn't pass what to cache.

I'd revert the whole thing back if needed, but overall it was very hard for me to follow.
I kinda understand now what it does, but the flow was hard to grasp.

I assume that it was to make sure that the result is cached ASAP, without getting the migration itself.
I'll try to implement this without this refactor.

Reply:
I agree that the current flow is a bit hard to follow, but I think we ran into numerous race conditions when debugging w/o using promises in the cache. 

I'd recommend falling back on the prev impl for these